juicy-html.html

```diff
@@ -38,194 +38,203 @@
             }
         }
         /**
-        * Checks whether a value is null or undefined
-        */
+         * Checks whether a value is null or undefined
+         */
         function nullOrUndefined(v) {
             return v === null || v === undefined;
         }
 
-        var JuicyHTMLPrototype = Object.create((HTMLTemplateElement || HTMLElement).prototype);
         var isSafari = navigator.vendor && navigator.vendor.indexOf("Apple") > -1 && navigator.userAgent && !navigator.userAgent.match("CriOS");
 
-        JuicyHTMLPrototype.createdCallback = function () {
-            var model = null; // XXX(tomalec): || this.model ? // to consider for https://github.com/Juicy/juicy-html/issues/30
-            Object.defineProperties(this, {
-                'model': {
-                    set: function (newValue) {
-                        model = newValue;
-                        this.attachModel(newValue);
-                    },
-                    get: function () {
-                        return model;
-                    }
-                },
-                'href': {
-                    set: function (newValue) {
-                        if (nullOrUndefined(newValue)) {
-                            this.removeAttribute('href');
-                        } else {
-                            this.setAttribute('href', newValue);
-                        }
-                    },
-                    get: function () {
-                        return this.getAttribute('href');
-                    }
-                },
-                'html': {
-                    set: function (newValue) {
-                        if (nullOrUndefined(newValue)) {
-                            this.removeAttribute('html');
-                        } else {
-                            this.setAttribute('html', newValue);
+        class JuicyHTMLElement extends HTMLElement {
+            static get observedAttributes() {
+                return ['model', 'href', 'html', 'content'];
+            }
+            constructor(self) {
+                self = super(self);
```

Comment:
Wasn't this line a `d-r-e` walkaround?

Reply:
The super call is required by spec, passing an argument is the requirement of `dre`, I'd keep it for the users of dre.